### ISSUE TITLE
Error when joining server

### ISSUE THREAD

**Issue by KrifixYT** [ESX-Org/es_extended#244]
Error running call reference function for resource esx_identity: citizen:/scripting/lua/scheduler.lua:403: @esx_identity/server/main.lua:11: attempt to index a nil value (field '?')
stack traceback:
        @esx_identity/server/main.lua:11: in upvalue 'ref'
        citizen:/scripting/lua/scheduler.lua:389: in function <citizen:/scripting/lua/scheduler.lua:388>
        [C]: in function 'xpcall'
        citizen:/scripting/lua/scheduler.lua:388: in function <citizen:/scripting/lua/scheduler.lua:387>
stack traceback:
        [C]: in function 'error'
        citizen:/scripting/lua/scheduler.lua:403: in function <citizen:/scripting/lua/scheduler.lua:372>
Error: Unhandled error in timer: Error: BUFFER_SHORTAGE
Error: BUFFER_SHORTAGE
    at n.e [as reserve] (citizen:/scripting/v8/msgpack.js:29:12766)
    at h (citizen:/scripting/v8/msgpack.js:29:15761)
    at n.r [as decode] (citizen:/scripting/v8/msgpack.js:29:13908)
    at n.fetch (citizen:/scripting/v8/msgpack.js:29:6972)
    at n.u [as read] (citizen:/scripting/v8/msgpack.js:29:12076)
    at Object.n [as decode] (citizen:/scripting/v8/msgpack.js:29:7097)
    at unpack (citizen:/scripting/v8/main.js:20:33)
    at citizen:/scripting/v8/main.js:51:20
    at setImmediate (mysql-async.js:4962:9)
    at Object.callback (citizen:/scripting/v8/timer.js:96:21)
Error running call reference function for resource es_extended: citizen:/scripting/lua/scheduler.lua:403: @es_extended/server/main.lua:133: attempt to index a nil value (field '?')
stack traceback:
        @es_extended/server/main.lua:133: in upvalue 'ref'
        citizen:/scripting/lua/scheduler.lua:389: in function <citizen:/scripting/lua/scheduler.lua:388>
        [C]: in function 'xpcall'
        citizen:/scripting/lua/scheduler.lua:388: in function <citizen:/scripting/lua/scheduler.lua:387>
stack traceback:
        [C]: in function 'error'
        citizen:/scripting/lua/scheduler.lua:403: in function <citizen:/scripting/lua/scheduler.lua:372>
Error: Unhandled error in timer: Error: BUFFER_SHORTAGE
Error: BUFFER_SHORTAGE
    at n.e [as reserve] (citizen:/scripting/v8/msgpack.js:29:12766)
    at h (citizen:/scripting/v8/msgpack.js:29:15761)
    at n.r [as decode] (citizen:/scripting/v8/msgpack.js:29:13908)
    at n.fetch (citizen:/scripting/v8/msgpack.js:29:6972)
    at n.u [as read] (citizen:/scripting/v8/msgpack.js:29:12076)
    at Object.n [as decode] (citizen:/scripting/v8/msgpack.js:29:7097)
    at unpack (citizen:/scripting/v8/main.js:20:33)
    at citizen:/scripting/v8/main.js:51:20
    at setImmediate (mysql-async.js:4962:9)
    at Object.callback (citizen:/scripting/v8/timer.js:96:21)
Error: (node:20092) UnhandledPromiseRejectionWarning: TypeError: Cannot convert undefined or null to object
    at Function.values (<anonymous>)
    at execute.then (mysql-async.js:5002:48)
Error: (node:20092) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
Error: (node:20092) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by Dpatau** [comment]
im also getting the same errors but also with esx_skin esx_properties esx_phone esx_status

anyone know of a fix

**Comment by KrifixYT** [comment]
*fixed*
rebuilded database one more time

**Comment by KrifixYT** [comment]
*fixed*

**Comment by ElPumpo** [comment]
that's esx_identity?!

**Comment by rgfirefly24** [comment]
So this is what I found and I would like to ask the developer why it was done this way.  FK References are based on the NAME field instead of the ID field between user -> items.  (user_inventory table)  What I saw is that when it attempted to pull a players inventory it was a null join to the item table which meant that the Label field was NULL causing the NIL error as seen above.   

So to the developer,  Why use the name field as a FK Reference?  I get that you would expect that to not change; however, I have seen multiple instances where people have changed a name for an item, job, ect  because it was either misspelled, in a Foreign language, or they wanted it to show different.  I.E.  CreditCard vs Credit Card.  If you used the ID field as the FK reference it would eliminate the chance that people making changes would cause the errors.  It's probably a big task at this point to fix as there are a lot of different tables that work this way.  

So my suggestion is to Just deal with the Names as they are.  That doesn't explain the fresh install issue, but even for you guys doing Fresh installs.  Run this query and see if fixing the results (if there are any) fixes your issue: 

SELECT DISTINCT user_inventory.* FROM user_inventory
LEFT JOIN items 
	ON user_inventory.item = items.NAME
WHERE items.name IS NULL
